src/compiler/compiler.ml

```diff
@@ -270,35 +270,58 @@ let check_defines com =
 	end
 
 (** Creates the typer context and types [classes] into it. *)
-let do_type ctx tctx actx =
-	let com = tctx.Typecore.com in
+let do_type ctx mctx actx display_file_dot_path =
+	let com = ctx.com in
 	let t = Timer.timer ["typing"] in
 	let cs = com.cs in
 	CommonCache.maybe_add_context_sign cs com "before_init_macros";
+	let macro_cache_enabled = !MacroContext.macro_enable_cache in
+	MacroContext.macro_enable_cache := true;
 	com.stage <- CInitMacrosStart;
-	List.iter (MacroContext.call_init_macro tctx) (List.rev actx.config_macros);
+	ServerMessage.compiler_stage com;
+
+	let mctx = List.fold_left (fun mctx path ->
```

Comment:
Why do we need a fold here? `mctx` should always be the same, right?

Reply:
Well the fold looks even weirder now that some things are gone, but it can still change _once_ since it may be defined or not initially depending on custom target.

Reply:
I guess we should just always initialize the macro context instead of playing this "is it already initialized or not" game. But I can look into cleaning that up later.

Reply:
Some tests currently rely on no macro context being created to check if things went correctly.
I think this can be cleaned up a bit while still not create macro context needlessly.